contrib/server/operations/pbench-upload-results.py

```diff
@@ -0,0 +1,270 @@
+        "--verify", "-v", dest="verify", action="store_true", help="Show progress"
+    )
+    parsed = parser.parse_args()
+
+    # Get basic configuration
+    host = socket.gethostname()
+    v = Path("/opt/pbench-server/VERSION")
+    s = Path("/opt/pbench-server/SHA1")
+    version = v.read_text().strip() if v.exists() else "unknown"
+    sha1 = s.read_text().strip() if s.exists() else "unknown"
+    metadata = [
+        f"global.server.legacy.hostname:{host}",
+        f"global.server.legacy.sha1:{sha1}",
+        f"global.server.legacy.version:{version}",
+    ]
+    if parsed.metadata:
+        metadata.extend(parsed.metadata)
+
+    since: Optional[datetime.datetime] = None
+    before: Optional[datetime.datetime] = None
+
+    since_ts: Optional[float] = None
+    before_ts: Optional[float] = None
+
+    if parsed.since:
+        since = date_parser.parse(parsed.since)
+        since_ts = since.timestamp()
+
+    if parsed.before:
+        before = date_parser.parse(parsed.before)
+        before_ts = before.timestamp()
+
+    if since and before:
+        when = f" (from {since:%Y-%m-%d %H:%M} to {before:%Y-%m-%d %H:%M})"
+    elif since:
+        when = f" (since {since:%Y-%m-%d %H:%M})"
+    elif before:
+        when = f" (before {before:%Y-%m-%d %H:%M})"
+    else:
+        when = ""
+
+    if parsed.tarball.is_dir():
+        what = f"DIRECTORY {parsed.tarball}"
+    else:
+        what = f"TARBALL {parsed.tarball}"
+
+    print(f"{what}{when} -> SERVER {parsed.server}")
+
+    checkpoint: Optional[Path] = None
+    processed: list[str] = []
+    if parsed.checkpoint:
+        checkpoint = parsed.checkpoint
+        if parsed.verify:
+            print(f"Processing checkpoint state from {checkpoint}...")
+        if checkpoint.exists():
+            processed = checkpoint.read_text().splitlines()
+            if parsed.verify:
+                print(f"[CPT] done {len(processed)}")
+        if parsed.verify:
+            print(
+                f"Finished processing checkpoint data: {len(processed)} checkpointed files"
+            )
+
+    if parsed.verify:
+        print("Identifying target tarballs")
+    skipped = 0
+    early = 0
+    late = 0
+    if parsed.tarball.is_dir():
+        pool = parsed.tarball.glob("**/*.tar.xz")
+        which = []
+        for t in pool:
+            if not t.is_file():
+                continue
+            date = t.stat().st_mtime
+            if since_ts and date < since_ts:
+                early += 1
+                continue
+            if before_ts and date >= before_ts:
+                late += 1
+                continue
+            if str(t) in processed:
+                skipped += 1
+                if parsed.verify:
+                    print(f"[CPT] skip {t}")
+                continue
+            which.append(t)
+    elif parsed.tarball.is_file():
+        which = [parsed.tarball]
+    else:
+        print(f"Path {parsed.tarball} doesn't exist", file=sys.stderr)
```

Comment:
Instead of "doesn't exist", perhaps "is not a valid file or directory".

Reply:
If I don't like it, it doesn't exist! Technicalities ... 😆